miner/stratum/src/traits.rs

```diff
@@ -55,7 +55,7 @@ pub trait JobDispatcher: Send + Sync {
 /// Interface that can handle requests to push job for workers
 pub trait PushWorkHandler: Send + Sync {
 	/// push the same work package for all workers (`payload`: json of pow-specific set of work specification)
-	fn push_work_all(&self, payload: String) -> Result<(), Error>;
```

Comment:
The trait looks a bit weird after this change because now `push_work_all` is `infallible` and `push_work` can fail.

I do realize it is because we don't want to `short-circuit` if one message fails when we need to send `x more` but still I guess it would be possible to keep all failed messages in a buffer and return `Err` if !buffer.is_empty()

Reply:
> The trait looks a bit weird

It does, but it's better to get the weirdness "in your face" than hidden behind a comfy `Result` that is always `Ok`.

> it would be possible to keep all failed messages in a buffer

Yeah, and while that would be more satisfying I still wonder what use calling code can make with such an error? Honest question: would it be useful?

Reply:
> Yeah, and while that would be more satisfying I still wonder what use calling code can make with such an error? Honest question: would it be useful?

I don't think so currently it is only called from [here](https://github.com/paritytech/parity-ethereum/blob/master/ethcore/src/miner/stratum.rs#L226) AFAIK and would require more changes then.

Also, seems like `push_work` is not used except in the tests.

We can merge this and investigate my points above separately